mindsdb_sql/planner/query_planner.py

```diff
@@ -1255,6 +1333,164 @@ def plan_union(self, query):
 
         return self.plan.add_step(UnionStep(left=query1.result, right=query2.result, unique=query.unique))
 
+
+    def plan_select_knowledege_base(self, query):
```

Comment:
wouldn't be easier to have this (and insert) logic in mindsdb in knowledge base controller ?
mindsdb executer receives select from table and if this is KB  route it to KB controller. 
KB controller calls embedding model or vector base  when it decides to do it

Reply:
That's a valid point. I thought about it for a while. 
But since KB is a wrapper around vector database and embedding model, I think it might be a good idea to do the "query rewriting / translating", compared to a dedicated controller that handles the execution. 

It also allows us to reuse all the nice things we built around tables and integrations and will support composite and nested queries by default (since we are not building new things but rather just doing translation of queries).

But I do agree this approach seems a bit intrusive.

Reply:
Cons of handling KB logic in planner:
- Sql transformation could be not save, for example 'knowledge_base join model' doesn't work well, and maybe other combinations should checked
- spreading logic over different places instead of encapsulation in controller. for now KB is two steps execution: calling the embedding model and selecting from vector db. But it could become more complex after starting of using it, for example: 
  - KB would use internal vector db not exposed to user
  - additional check for inserted rows or query string. 
- not back compatible update of logic might require little bit complex release process (mindsdb and sql) 

For KB controller I don't see cons yet. I think KB table will be used as table in fetchdataframe step and we already have entity working in similar way: a view  
Maybe I miss something

Reply:
So the suggestion is to 
- Leave the planner as is
- SELECT/INSERT/UPDATE operations to KB will be parsed as operations to a normal database table (need a bit of confirmation)
- In the executor, do an extra check to differentiate operations to tables vs KB
- Route operations to KB to a KB controller (which I believe will implement SELECT/UPDATE/INSERT/DELETE operations correspondingly). 

@ea-rus